pkg/volume/persistent_disk/persistent_disk.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disk
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume"
+	"github.com/golang/glog"
+	"os"
+)
+
+type persistentDiskPluginInterface interface {
+	Name() string
+	GetHost() volume.VolumeHost
+}
+
+type PersistentDiskPlugin struct {
+	PluginName string
+	Host       volume.VolumeHost
+}
+
+func (pd *PersistentDiskPlugin) Name() string {
+	return pd.PluginName
+}
+
+func (pd *PersistentDiskPlugin) GetHost() volume.VolumeHost {
+	return pd.Host
+}
+
+func (plugin *PersistentDiskPlugin) Init(host volume.VolumeHost) {
+	plugin.Host = host
+}
+
+type persistentDiskInterface interface {
+	// SetUp attaches the disk and bind mounts to the volume path.
+	SetUp() error
+	SetUpAt(path string) error
+
+	// Unmounts the bind mount, and detaches the disk only if the PD
+	// resource was the last reference to that disk on the kubelet.
+	TearDown()
+	TearDownAt() error
+
+	GetPath() string
+}
+
+type PersistentDisk struct {
+	Plugin  persistentDiskPluginInterface
+	PodUID  types.UID
+	VolName string
+
+	Mounter persistentDiskMounter
+}
```

Comment:
I think this overloads the term "Persistent" wrongly.  I have to reason through this refactor more deeply to understand what you're trying to do, but all the common code belongs to "network" volumes that involve mounting.  They may or may not be a PersistentVolume.